tests/unit/components/test_braveadblock.py

```diff
@@ -0,0 +1,403 @@
+        assert_urls(ad_blocker, NOT_OKAY_URLS, True)
+        assert_urls(ad_blocker, OKAY_URLS, False)
+
+        # After setting the ad blocking lists to None, the ads should still be
+        # blocked, since we haven't run `:brave-adblock-update`.
+        config_stub.val.content.blocking.adblock.lists = None
+        assert_urls(ad_blocker, NOT_OKAY_URLS, True)
+        assert_urls(ad_blocker, OKAY_URLS, False)
+
+        # After updating the adblocker, nothing should be blocked, since we set
+        # the blocklist to None.
+        downloads = ad_blocker.adblock_update()
+        while downloads._in_progress:
+            current_download = downloads._in_progress[0]
+            with caplog.at_level(logging.ERROR):
+                current_download.successful = True
+                current_download.finished.emit()
+        assert_none_blocked(ad_blocker)
+
+
+def test_whitelist_on_dataset(config_stub, easylist_easyprivacy):
+    config_stub.val.content.blocking.adblock.lists = easylist_easyprivacy
+    config_stub.val.content.blocking.enabled = True
+    config_stub.val.content.blocking.whitelist = None
+
+    def assert_whitelisted(url, source_url, resource_type):
+        config_stub.val.content.blocking.whitelist = None
+        assert not blockutils.is_whitelisted_url(url)
+        config_stub.val.content.blocking.whitelist = []
+        assert not blockutils.is_whitelisted_url(url)
+        whitelist_url = url.toString(QUrl.RemovePath) + "/*"
+        config_stub.val.content.blocking.whitelist = [whitelist_url]
+        assert blockutils.is_whitelisted_url(url)
+
+    run_function_on_dataset(assert_whitelisted)
+
+
+def test_update_easylist_easyprivacy_directory(
+    ad_blocker, config_stub, easylist_easyprivacy_both, caplog
+):
+    # This directory should contain two text files, one for easylist, another
+    # for easyprivacy.
+    lists_directory = easylist_easyprivacy_both[1]
+
+    config_stub.val.content.blocking.adblock.lists = [
+        QUrl.fromLocalFile(str(lists_directory)).toString()
+    ]
+    config_stub.val.content.blocking.enabled = True
+    config_stub.val.content.blocking.whitelist = None
+
+    with caplog.at_level(logging.INFO):
+        ad_blocker.adblock_update()
+        assert_only_one_success_message(caplog.messages)
+        assert (
+            caplog.messages[-1]
+            == "braveadblock: Filters successfully read from 2 sources"
+        )
+    assert_urls(ad_blocker, NOT_OKAY_URLS, True)
+    assert_urls(ad_blocker, OKAY_URLS, False)
+
+
+def test_update_empty_directory_blocklist(ad_blocker, config_stub, empty_dir, caplog):
+    tmpdir_url = QUrl.fromLocalFile(empty_dir).toString()
+    config_stub.val.content.blocking.adblock.lists = [tmpdir_url]
+    config_stub.val.content.blocking.enabled = True
+    config_stub.val.content.blocking.whitelist = None
+
+    # The temporary directory we created should be empty
+    assert len(os.listdir(empty_dir)) == 0
```

Comment:
In this case I think it's more readable to use `len`, since some readers may not know the return type of `os.listdir`. Using `len` gives the hint that it's a container of some sort.